src/util/pushSender.ts

```diff
@@ -72,35 +77,37 @@ export function makePushSender(connection: ServerScope): PushSender {
     return sender
   }
 
-  async function sendRaw(
-    apiKey: string,
-    tokens: Set<string>,
-    message: PushMessage
-  ): Promise<PushResult> {
-    const { title = '', body = '', data = {} } = message
-
-    const failure = {
-      successCount: 0,
-      failureCount: tokens.size
-    }
-
-    const sender = await getSender(apiKey)
-    if (sender == null) return failure
-
-    const response = await sender
-      .sendMulticast({
-        data,
-        notification: { title, body },
-        tokens: [...tokens]
-      })
-      .catch(() => failure)
-
-    successCounter.inc(response.successCount)
-    failureCounter.inc(response.failureCount)
-    return response
-  }
+  const instance: PushSender = {
+    async sendRaw(
```

Comment:
Optional: Another approach that works well is to leave `sendRaw` and `send` as loose functions, and then do `return {send, sendRaw}` at the end. That way, they can call each other by name, and you have less indentation overall.

Reply:
I do like seeing `instance.sendRaw` over `sendRaw` to distinguish a private invocation from a public one. Also, you can see which functions are public by them being indented in the instance declaration.